util-images/network/netperfbenchmark/pkg/util/util.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package util contains utility methods for worker
+package util
+
+import (
+	"errors"
+	"math"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"k8s.io/klog"
+)
+
+// Protocols supported
+const (
+	ProtocolTCP  = "TCP"
+	ProtocolUDP  = "UDP"
+	ProtocolHTTP = "HTTP"
+)
+
+// Number of metrics for each protocol
+const (
+	tcpMtrCnt   = 2
+	udpMtrCnt   = 11
+	httpMetrCnt = 12
+)
+
+// Iperf results vary from protocol,required for proper parsing
+const (
+	zeroFmtTCP         = "0.0"
+	fractionZeroFmtTCP = ".0"
+	zeroFmtUDP         = "0.00"
+	fractionZeroFmtUDP = ".00"
+)
+
+// Function to be applied for each metric for aggregation
+var (
+	iperfUDPFn = []string{"", "", "", "Sum", "Sum", "Sum", "Sum", "Sum", "Avg", "Avg", "Min", "Max", "Avg", "Sum"}
+	iperfTCPFn = []string{"", "", "", "Sum", "Avg"}
+)
+
+// ParseResult parses the response received for each protocol type
+func ParseResult(protType string, result []string, testDuration string) ([]float64, error) {
+	klog.Info("Parsing response for ", protType)
+	switch protType {
+	case ProtocolTCP:
+		return parseIperfResponse(result, testDuration, ProtocolTCP, tcpMtrCnt, zeroFmtTCP, fractionZeroFmtTCP, iperfTCPFn), nil
+	case ProtocolUDP:
+		return parseIperfResponse(result, testDuration, ProtocolUDP, udpMtrCnt, zeroFmtUDP, fractionZeroFmtUDP, iperfUDPFn), nil
+	case ProtocolHTTP:
+		return parseSiegeResponse(result), nil
+	default:
+		return nil, errors.New("invalid protocol:" + protType)
+	}
+}
+
+func parseIperfResponse(result []string, testDuration string, protocol string, metricCnt int, zeroFmt string, fractionZeroFmt string, operType []string) []float64 {
+	klog.Info("In parseIperf")
+	sumResult := make([]float64, 0, metricCnt)
+	unitReg := regexp.MustCompile(`%|\[\s+|\]\s+|KBytes\s+|KBytes/sec\s*|sec\s+|pps\s*|ms\s+|/|\(|\)\s+`)
+	mulSpaceReg := regexp.MustCompile(`\s+`)
+	hypSpcReg := regexp.MustCompile(`\-\s+`)
+	cnt := 0
+	sessionID := make(map[string]bool)
+	for _, op := range result {
+		klog.Info(op)
```

Comment:
remove - let's not log everything

Same for other similar logs below

Reply:
Modified.